python/ray/tune/experiment/trial.py

```diff
@@ -186,6 +191,58 @@ def _create_unique_logdir_name(root: str, relative_logdir: str) -> str:
     return relative_logdir
 
 
+def _noop_logger_creator(
+    config: Dict[str, Any], logdir: str, should_chdir: bool = True
+):
+    # Upon remote process setup, record the actor's original working dir before
+    # changing to the Tune logdir
+    os.environ.setdefault("TUNE_ORIG_WORKING_DIR", os.getcwd())
+
+    os.makedirs(logdir, exist_ok=True)
+    if should_chdir:
+        # Set the working dir to the trial directory in the remote process,
+        # for user file writes
+        if not ray._private.worker._mode() == ray._private.worker.LOCAL_MODE:
+            os.chdir(logdir)
+    return NoopLogger(config, logdir)
+
+
+def _get_trainable_kwargs(
+    trial: "Trial",
+    additional_kwargs: Optional[Dict[str, Any]] = None,
+    should_chdir: bool = False,
+) -> Dict[str, Any]:
+    trial.init_logdir()
```

Comment:
Can we move this back out to `setup_remote_runner`? Feels a bit weird to do in this helper fn.

Reply:
The main motivation behind this PR is to move this out of the TrialExecutor. The trainable kwargs only depend on trial properties, so there is no reason why this should be kept in the executor. For a bit more context, we will need to setup trainable kwargs in other locations (e.g. for unit testing, and in the new trainable execution path), which would otherwise duplicate code.

Reply:
Oh, I mean the `trial.init_logdir()` specifically. I think it's actually fine, since this is just a noop if it already exists, and we do require `trial.logdir` later.